test/integration_tests.jl

```diff
@@ -4,27 +4,43 @@
 # You should add the PAT as an encrypted Travis CI environment variable named `BCBI_TEST_USER_GITHUB_TOKEN`.
 # You should generate an SSH deploy key. Upload the public key as a deploy key for the testing repo.
 # You should make the private key an encrypted Travis CI environment variable named `COMPATHELPER_PRIV`. Probably you will want to Base64-encode it.
+const GLOBAL_PR_TITLE_PREFIX = Random.randstring(8)
+const GITHUB = "GITHUB"
+const GITLAB = "GITLAB"
 
-@testset "GitHub Integration Test" begin
-    GLOBAL_PR_TITLE_PREFIX = Random.randstring(8)
-    COMPATHELPER_INTEGRATION_TEST_REPO = ENV["COMPATHELPER_INTEGRATION_TEST_REPO"]
-    TEST_USER_GITHUB_TOKEN = ENV["BCBI_TEST_USER_GITHUB_TOKEN"]
+function get_api(service_name, personal_access_token)
+    if service_name == GITHUB
+        token = GitHub.Token(personal_access_token)
+        return GitHub.GitHubAPI(; token=token)
+    elseif service_name == GITLAB
+        token = GitLab.PersonalAccessToken(personal_access_token)
+        return GitLab.GitLabAPI(; token=token)
+    end
+end
 
-    api = GitHub.GitHubAPI(; token=GitHub.Token(TEST_USER_GITHUB_TOKEN))
-    user, _ = GitForge.get_user(api)
-    repo, _ = GitForge.get_repo(api, COMPATHELPER_INTEGRATION_TEST_REPO)
+function ci_config(service_name)
+    if service_name == GITHUB
+        return CompatHelper.GitHubActions()
+    elseif service_name == GITLAB
+        return CompatHelper.GitLabCI()
+    end
+end
```

Comment:
You might be able to use `auto_detect_ci_service` here instead? It relies on ENV variables being set so maybe not quite a direct replacement. 

https://github.com/JuliaRegistries/CompatHelper.jl/blob/v3-dev/src/utilities/ci.jl#L56-L64